### ISSUE TITLE
Add `to_b` and `to_h` to `JS::Object`

### ISSUE THREAD

**Issue by terrablue** [ruby/ruby.wasm#346]
I would like to suggest adding `to_b` and `to_h` to `JS::Object`, to convert into a ruby boolean respective hash.

to_b → boolean
Converts `self` to `true` or `false`:
```rb
JS.eval("return true").to_b        # => true
JS.eval("return false").to_b       # => false
```
to_h → hash
Converts `self` to a Hash:
```rb
JS.eval("return {foo: 'bar'}").to_h         # => { "foo" => "bar" }
```

**Comment by kateinoigakukun** [comment]
For `to_b`, I intentionally didn't add it. See https://github.com/ruby/ruby.wasm/pull/247 for details.

**Comment by terrablue** [comment]
I understand. So in that case I can implement it myself for my use case. What about `to_h`? It's already used in this meaning in ruby, no?

**Comment by kateinoigakukun** [comment]
Right, `to_h` sounds reasonable to me. I think that would be something like https://github.com/ruby/ruby.wasm/pull/270. Note that the conversion should be shallow, not recursive.
